.github/workflows/e2e.yml

```
@@ -156,10 +156,50 @@ jobs:
       - name: Run e2e test
         run: timeout 2400 bats "e2e/$E2E_TEST"
 
+  ui_test:
+    runs-on: ${{ matrix.os }}
+    needs: [build_dfx]
+    strategy:
+      matrix:
+        os: [macos-11, ubuntu-20.04]
+        rust: ['1.60.0']
+    steps:
+      - name: Checking out repo
+        uses: actions/checkout@v2
```

let's mimic `.jobs.test` more closely
```suggestion
        uses: actions/checkout@v3
```